src/Game.java

```diff
@@ -52,29 +54,48 @@ private static void TravelLoop(Player player, Landmark destination){
             } catch (InterruptedException e) {
                 throw new RuntimeException(e);
             }
+            currentTravelMiles += travelDistance;
             player.setMilesTraveled(player.getMilesTraveled() + travelDistance);
         }
-        Utils.println("arrived", destination.name);
+        Utils.println("arrived", finalDestination.getName());
     }
 
     private static void GameLoop(Player player){
-        Iterator i = TRAILS.iterator();
-        while(i.hasNext()) {
-            TravelLoop(player, (Landmark) i.next());
-        }
+        Node current = TRAILS.head;
+        Node next;
+            while(current.hasNext()){
+                //determine next destination
+                Destination finalDestination;
+                if(current.altNext() != null){
+                    //there are multiple options
+                    Destination d1 = current.getData().points[0];
+                    Destination d2 = current.getData().points[1];
+
+                    //display fork message
+                    Utils.println("fork",d1.getName(),d1.getDistance(), d2.getName(), d2.getDistance());
+
+                    //get user choice and set destination landmark
+                    finalDestination = Utils.choice(current.getData().points);
+                    if (finalDestination.equals(d1)){
+                        next = current.next();
+                    } else {
+                        next = current.altNext();
+
+                    }
+                } else {
+                    //no option, set next destination
+                    finalDestination = current.getData().points[0];
+                    finalDestination.setName(current.next().getData().name);
+                    next = current.next();
+                }
+                TravelLoop(player, current, next, finalDestination );
+                System.out.println("Total Miles Traveled: " + player.getMilesTraveled());
```

Comment:
The total miles travelled is incorrect. The total should not go over the actual distance to the destination. For example, the first destination is `102 miles` away, but outputs `20 * 6 = 120 miles`